File: app/src/main/java/co/edu/unicauca/dopaminah/domain/model/AppLimitGoal.kt
package co.edu.unicauca.dopaminah.domain.model

/**
 * Represents a user-defined usage limit goal.
 * - TOTAL_DAILY: limits total screen time per day (maxTimeMillis)
 * - APP_LIMIT: limits a specific app's usage (maxTimeMillis, packageName + appDisplayName)
 * - UNLOCK_LIMIT: limits number of device unlocks (maxUnlocks)
 */
data class AppLimitGoal(
    val id: Int = 0,
    val goalType: String,             // "TOTAL_DAILY" | "APP_LIMIT" | "UNLOCK_LIMIT"
    val packageName: String = "",     // Only used for APP_LIMIT
    val appDisplayName: String = "",  // Human-readable name of the target app
    val maxTimeMillis: Long = 0L,     // Time limit in milliseconds (TOTAL_DAILY / APP_LIMIT)
    val maxUnlocks: Int = 0,          // Unlock limit (UNLOCK_LIMIT)
    val currentStreak: Int = 0        // Days user met the goal
)

File: DopamiNah/Domain/Models/AppLimitGoal.swift
import Foundation
import SwiftData

@Model
final class AppLimitGoal {
    @Attribute(.unique) var id: UUID
    var goalType: String
    var packageName: String
    var appDisplayName: String
    var maxTimeMillis: Int64
    var maxUnlocks: Int
    var currentStreak: Int
    var lastStreakUpdateDate: Date?

    init(
        id: UUID = UUID(),
        goalType: String,
        packageName: String = "",
        appDisplayName: String = "",
        maxTimeMillis: Int64 = 0,
        maxUnlocks: Int = 0,
        currentStreak: Int = 0,
        lastStreakUpdateDate: Date? = nil
    ) {
        self.id = id
        self.goalType = goalType
        self.packageName = packageName
        self.appDisplayName = appDisplayName
        self.maxTimeMillis = maxTimeMillis
        self.maxUnlocks = maxUnlocks
        self.currentStreak = currentStreak
        self.lastStreakUpdateDate = lastStreakUpdateDate
    }
}

struct GoalType {
    static let totalDaily = "TOTAL_DAILY"
    static let appLimit = "APP_LIMIT"
    static let unlockLimit = "UNLOCK_LIMIT"
}

struct AppLimitCardInfo: Identifiable {
    let id = UUID()
    let packageName: String
    let appName: String
    let timeUsedMs: Int64
    let timeLimitMs: Int64
}
